musicbrainzngs/musicbrainz.py

```diff
@@ -587,6 +492,28 @@ def set_format(fmt="xml"):
         raise ValueError("invalid format: %s" % fmt)
 
 
+def _safe_read(request):
+    """
+    :param request:
+    """
+    # Make request (with retries).
+    with requests.Session() as session:
+        adapter = requests.adapters.HTTPAdapter(max_retries=_retry)
+        session.mount('http://', adapter)
+        session.mount('https://', adapter)
+        try:
+                resp = session.send(request.prepare(), allow_redirects=True)
+                resp.raise_for_status()
+                return resp
+        except requests.HTTPError as exc:
+                if exc.response.status_code == 401:
+                        raise AuthenticationError(cause=exc)
+                raise ResponseError(cause=exc)
+        except requests.RetryError as exc:
+                raise NetworkError(cause=exc)
```

Comment:
I could have this wrong, but it looks on my machine like `RetryError` is in the `requests.exceptions` module, not just in `requests`:

```py
>>> requests.__version__
'2.9.1'
>>> requests.RetryError
Traceback (most recent call last):
  File "<input>", line 1, in <module>
    requests.RetryError
AttributeError: 'module' object has no attribute 'RetryError'
>>> requests.exceptions.RetryError
<class 'requests.exceptions.RetryError'>
```

Also, I noticed that `RetryError` inherits from `RequestException`, so it would have the same effect to just fall through to the next `except` handler.